src/core/tests/type_prop/is_nan.cpp

```diff
@@ -3,35 +3,32 @@
 //
 
 #include "gtest/gtest.h"
-#include "openvino/opsets/opset9.hpp"
 #include "openvino/opsets/opset10.hpp"
 #include "util/type_prop.hpp"
 
 using namespace std;
 using namespace ov;
 
 TEST(type_prop, isnan_output_shape) {
-    const auto data = make_shared<opset9::Parameter>(ov::element::f16, Shape{4, 2});
+    const auto data = make_shared<opset10::Parameter>(ov::element::f16, Shape{4, 2});
     const auto isnan = make_shared<opset10::IsNaN>(data);
 
     EXPECT_TRUE(isnan->get_output_partial_shape(0).same_scheme(PartialShape{4, 2}))
         << "The output shape of IsNaN is incorrect";
-    ASSERT_EQ(isnan->get_shape(), (Shape{4, 2}))
-        << "The output shape of IsNaN is incorrect";
+    ASSERT_EQ(isnan->get_shape(), (Shape{4, 2})) << "The output shape of IsNaN is incorrect";
 }
 
 TEST(type_prop, isnan_sample_dynamic_batch) {
-    const auto data = make_shared<opset9::Parameter>(ov::element::f16, PartialShape{Dimension::dynamic(), 21, 37});
+    const auto data = make_shared<opset10::Parameter>(ov::element::f16, PartialShape{Dimension::dynamic(), 21, 37});
     const auto isnan = make_shared<opset10::IsNaN>(data);
 
     EXPECT_TRUE(isnan->get_output_partial_shape(0).same_scheme(PartialShape{Dimension::dynamic(), 21, 37}))
```

Comment:
Use: 
```EXPECT_EQ(isnan->get_output_partial_shape(0), PartialShape({...}))```

This gives more reliable check of expected shape.

Add some more compelx shape or dedicated test to propagate interval  dimension ```Dimension(2,4), Dimension(-1, 5)```, etc. and fully dynamic shape ```PartialShape::dynamic(), PartialShape::dynamic(5) etc.```

Reply:
done